### PR TITLE
Create Renderer class

### PR DESCRIPTION
**Pull request by HugoN1**
### Planed changes :
- Create Renderer class

### Not Planned changes :
- Fix memory handling in Communication class
- Fix Game class constructor

_It has not been implemented with unit tests because Renderer class actually only contains getters and setters_

### CODE REVIEW

File: gui/src/client/ICommunication/Communication.hpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #pragma once
 #include "ICommunication.hpp"
 #include "../../../../includes/global.hpp"
-#include "../../data.hpp"
+#include "../../team.hpp"
```
Comment (AndreasLETANTER) [inline gui/src/client/ICommunication/Communication.hpp:11]: does the -Isrc in the makefile could help you remove the ../ ?

**Review by Rafouzou — APPROVED** [review]

File: gui/src/game/IGame/Game.cpp [hunk 2]
```diff
@@ -33,9 +34,9 @@ void Game::run()
...
 }
 
-std::shared_ptr<Terrain> Game::getMap() const
+std::shared_ptr<Data> Game::getData() const
 {
-    return m_map;
+    return m_data;
 }
 
 std::pair<int, int> Game::getScale() const
```
Comment (lbarreteau) [inline gui/src/game/IGame/Game.cpp:42]: Unsigned int

File: gui/src/game/IGame/Game.hpp [hunk 3]
```diff
@@ -12,17 +12,17 @@
...
-        Game();
+        Game(std::shared_ptr<Data> t_data);
         ~Game() override = default;
-        void setMap(std::shared_ptr<Terrain> &t_map) override;
+        void setData(std::shared_ptr<Data> t_data) override;
         void run() override;
-        std::shared_ptr<Terrain> getMap() const override;
-        std::pair<int, int> getScale() const;
+        std::shared_ptr<Data> getData() const override;
+        std::pair<int, int> getScale() const override;
```
Comment (lbarreteau) [inline gui/src/game/IGame/Game.hpp:20]: Unsigned

File: gui/src/game/IGame/Game.hpp [hunk 4]
```diff
@@ -12,17 +12,17 @@
...
     private:
         std::pair<int, int> m_window_size;
         std::pair<int, int> m_scale;
```
Comment (lbarreteau) [inline gui/src/game/IGame/Game.hpp:23]: Unsigned int

File: gui/src/game/IGame/IGame.hpp [hunk 5]
```diff
@@ -6,14 +6,14 @@
...
 
 class IGame {
     public:
         virtual ~IGame() = default;
-        virtual void setMap(std::shared_ptr<Terrain> &t_map) = 0;
+        virtual void setData(std::shared_ptr<Data> t_data) = 0;
         virtual void run() = 0;
-        virtual std::shared_ptr<Terrain> getMap() const = 0;
+        virtual std::shared_ptr<Data> getData() const = 0;
         virtual std::pair<int, int> getScale() const = 0;
```
Comment (lbarreteau) [inline gui/src/game/IGame/IGame.hpp:18]: Unsigned int

File: gui/src/game/IRenderer/IRenderer.hpp [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+class IRenderer {
+    public:
+        virtual ~IRenderer() = default;
+        virtual void setTexture(sf::Texture t_texture) = 0;
+        virtual void setSprite(sf::Sprite t_sprite) = 0;
+        virtual void setTextureRect(sf::IntRect t_intRect) = 0;
+        virtual sf::Texture getTexture(void) = 0;
+        virtual sf::IntRect getTextureRect(void) = 0;
+        virtual sf::Sprite getSprite(void) = 0;
+        virtual std::pair<int, int> getPos(void) = 0;
```
Comment (lbarreteau) [inline gui/src/game/IRenderer/IRenderer.hpp:20]: Unsigned int

File: gui/src/game/IRenderer/Renderer.cpp [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+{
+    return m_sprite.getTextureRect();
+}
+
+sf::Sprite Renderer::getSprite(void)
+{
+    return m_sprite;
+}
+
+std::pair<int, int> Renderer::getPos(void)
```
Comment (lbarreteau) [inline gui/src/game/IRenderer/Renderer.cpp:46]: Unsigned int

File: gui/src/game/IRenderer/Renderer.hpp [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    public:
+        Renderer(std::string t_textureFileName, std::pair<int, int> t_pos);
+        ~Renderer() override = default;
+        void setTexture(sf::Texture t_texture) override;
+        void setSprite(sf::Sprite t_sprite) override;
+        void setTextureRect(sf::IntRect t_intRect) override;
+        sf::Texture getTexture(void) override;
+        sf::IntRect getTextureRect(void) override;
+        sf::Sprite getSprite(void) override;
+        std::pair<int, int> getPos(void) override;
```
Comment (lbarreteau) [inline gui/src/game/IRenderer/Renderer.hpp:21]: Unsigned int

File: gui/src/game/IRenderer/Renderer.hpp [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+** File description:
+** Renderer class header file
+*/
+
+#pragma once
+#include "IRenderer.hpp"
+
+class Renderer : public IRenderer {
+    public:
+        Renderer(std::string t_textureFileName, std::pair<int, int> t_pos);
```
Comment (lbarreteau) [inline gui/src/game/IRenderer/Renderer.hpp:13]: Unsigned int

File: gui/src/game/IRenderer/Renderer.hpp [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+        void setTextureRect(sf::IntRect t_intRect) override;
+        sf::Texture getTexture(void) override;
+        sf::IntRect getTextureRect(void) override;
+        sf::Sprite getSprite(void) override;
+        std::pair<int, int> getPos(void) override;
+    private:
+        sf::Texture m_texture;
+        sf::Sprite m_sprite;
+        sf::IntRect m_textureRect;
+        std::pair<int, int> m_pos;
```
Comment (lbarreteau) [inline gui/src/game/IRenderer/Renderer.hpp:26]: Unsigned int

**Review by lbarreteau — CHANGES_REQUESTED** [review]
Unsigned = non signé

**Review by lbarreteau — APPROVED** [review]
Clean 💎

**Pull request merged**